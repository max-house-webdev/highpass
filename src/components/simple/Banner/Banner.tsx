import { StackProps, VStack } from "@chakra-ui/react";

export interface IBannerProps extends StackProps {}

export function Banner(props: IBannerProps) {
  const { children, ...rest } = props;

  return (
    <VStack
      {...rest}
      justifyContent={"flex-end"}
      alignItems={"flex-start"}
      minHeight={"422px"}
      paddingInlineStart={7}
      paddingInlineEnd={2}
      paddingBlockStart={2}
      paddingBlockEnd={7}
      bgRepeat={"no-repeat"}
      bgSize={"cover"}
    >
      {children}
    </VStack>
  );
}
