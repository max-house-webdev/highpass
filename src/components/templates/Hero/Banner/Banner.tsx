import { Text } from "@chakra-ui/react";

import { v4 as uuidv4 } from "uuid";

import { BannerStack } from "@components/atoms/simple/BannerStack";

import { BannerHeading } from "@components/atoms/simple/BannerHeading";
import { Hashtag } from "@components/atoms/simple/Hashtag";
import { LearnMore as BtnLearnMore } from "@components/atoms/ui/buttons/LearnMore";

import { content } from "@core/content";

export function Banner() {
  const { sections } = content;
  const { hero } = sections;
  const { banner } = hero;

  const { bgImage, btnTextContent, hashtag, heading, description } = banner;

  return (
    <BannerStack data-testid="Banner" bgImage={bgImage} marginBlockEnd={8}>
      <Hashtag textContent={hashtag} />

      <Text variant={"bannerDescription"}>{description}</Text>

      {heading.map((line) => (
        <BannerHeading key={uuidv4()}>{line}</BannerHeading>
      ))}

      <BtnLearnMore textContent={btnTextContent} marginBlockStart={2} />
    </BannerStack>
  );
}
