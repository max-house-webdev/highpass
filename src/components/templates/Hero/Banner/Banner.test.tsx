import React from "react";

import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { Banner } from "./Banner";

describe("<Banner />", () => {
  test("it should mount", () => {
    render(<Banner />);

    const banner = screen.getByTestId("Banner");

    expect(banner).toBeInTheDocument();
  });
});

test("<Banner /> renders correctly", () => {
  const tree = renderer.create(<Banner />).toJSON();
  expect(tree).toMatchSnapshot();
});
