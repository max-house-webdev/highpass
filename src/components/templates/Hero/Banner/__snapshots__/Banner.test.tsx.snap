// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> renders correctly 1`] = `
<div
  className="chakra-stack css-11ub2gy"
  data-testid="Banner"
>
  <p
    className="chakra-text css-0"
    data-testid="hashtag"
  >
    #мастеркласс
  </p>
  <p
    className="chakra-text css-0"
  >
    Мастер-класс от Игоря Робова
  </p>
  <h2
    className="chakra-heading css-0"
  >
    Неоновый
  </h2>
  <h2
    className="chakra-heading css-0"
  >
    фотосет 2.0
  </h2>
  <h2
    className="chakra-heading css-0"
  >
    7 июля 20:00
  </h2>
  <button
    className="chakra-button css-1lvhd6m"
    data-testid="LearnMore"
    onClick={[Function]}
    type="button"
  >
    Узнать больше
  </button>
</div>
`;
